harness/determined/exec/launch.py

```diff
@@ -0,0 +1,98 @@
+import copy
+import json
+import logging
+import re
+import subprocess
+import sys
+from typing import Dict
+
+import determined as det
+import determined.common
+from determined.common import constants, storage
+
+
+def match_legacy_trial_class(arg: str) -> bool:
+    trial_class_regex = re.compile("^[a-zA-Z0-9_.]+:[a-zA-Z0-9_]+$")
+    if trial_class_regex.match(arg):
+        return True
+    return False
+
+
+def launch(experiment_config: det.ExperimentConfig) -> int:
+    # If native is enabled, harness will load from native entrypoint command
+    if experiment_config.native_enabled():
+        from determined.exec import harness
+
+        return harness.main(train_entrypoint=None)
+
+    entrypoint = experiment_config.get_entrypoint()
+    slots_per_trial = experiment_config.slots_per_trial()
+
+    if not entrypoint:
+        logging.error("Entrypoint not specified in experiment config")
+        return 1
+
+    if isinstance(entrypoint, list):
+        entrypoint = " ".join(entrypoint)
```

Comment:
How do you distinguish between:
```yaml
entrypoint: "./my file with spaces"
```
and
```yaml
entrypoint: ["./my", "file", "with", "args"]
```
?

(there's a fundamental problem with your strategy here, can you name it?)

Reply:
ah, didn't account for spaces in commands. yeah, passing a string into shell makes it ambiguous as to whether you're looking for an executable with spaces in it or an executable with args passed in.

 i've made changes to pass in string or list as is to shell instead of doing any kind of parsing on commands.